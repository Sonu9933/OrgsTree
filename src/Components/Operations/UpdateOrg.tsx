import React, { useContext } from "react";
import { OrganizationContext } from "../Context/Context";
import { ActionTypes } from "../Context/ActionTypes";
import { Link } from "react-router-dom";
import { useCallback } from "react";
import { useDropzone } from "react-dropzone";

export const UpdateOrg = () => {
  const { state, dispatch } = useContext(OrganizationContext);

  const onDrop = useCallback(
    (acceptedFiles) => {
      if (acceptedFiles.length > 1) {
        dispatch({
          type: ActionTypes.ERROR,
          payload: {
            error: true,
          },
        });
      } else if (
        acceptedFiles[0].path
          .substr(acceptedFiles[0].path.lastIndexOf("\\") + 1)
          .split(".")[0] !== "json"
      ) {
        dispatch({
          type: ActionTypes.ERROR,
          payload: {
            error: true,
          },
        });
      } else {
        acceptedFiles.forEach((file: any) => {
          const reader = new FileReader();

          reader.onabort = () =>
            dispatch({
              type: ActionTypes.ERROR,
              payload: {
                error: true,
              },
            });
          reader.onerror = () =>
            dispatch({
              type: ActionTypes.ERROR,
              payload: {
                error: true,
              },
            });
          reader.onload = () => {
            const binaryStr = reader.result;
            dispatch({
              type: ActionTypes.UPDATE_JSONDATA,
              payload: { data: binaryStr, jsonUpdate: true },
            });
          };
          reader.readAsText(file);
        });
      }
    },
    [dispatch]
  );

  const { getRootProps, getInputProps } = useDropzone({ onDrop });
  const { jsonUpdate, error } = { ...state };
  return (
    <div className="container">
      <div className="row">
        <div className="col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto text-center form p-4">
          <h1 className="text-center display-4 py-2 text-truncate text-primary">
            <b className="b1">Upload JSON here!!</b>
          </h1>
        </div>
      </div>
      {!error ? (
        <>
          <div className="row">
            <div className="col-lg-8 col-md-8 col-sm-10 mx-auto text-center form p-4">
              {jsonUpdate ? (
                <h4 className="display-4 py-2 text-success">
                  JSON uploded successfully!! Click on GoTo Home link to update
                  organization chart
                </h4>
              ) : (
                <h4 className="display-4 py-2 border border-primary text-info rounded">
                  <div {...getRootProps()}>
                    <input {...getInputProps()} />
                    Drag 'n' drop some files here, or click to select files
                  </div>
                </h4>
              )}
            </div>
          </div>
          <div className="row">
            <div className="col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto text-center form p-4">
              Go back to Home
              <b>
                <Link to="/" className="Info link px-2">
                  Goto Home
                </Link>
              </b>
            </div>
          </div>
        </>
      ) : (
        <div className="row">
          <div className="col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto text-center form p-4">
            <b>
              Oops you cant upload more than 1 file or file is not in json
              format!!!
              <Link to="/" className="Danger link px-2">
                click here ToGo Home
              </Link>
            </b>
          </div>
        </div>
      )}
    </div>
  );
};
