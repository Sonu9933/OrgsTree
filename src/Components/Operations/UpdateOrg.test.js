import React from "react";
import { UpdateOrg } from "./UpdateOrg";
import { renderWithContext } from "../Test/TestHelper";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

describe("Update Org", () => {
  test("Should render UpdateOrg", () => {
    const component = renderWithContext(
      <BrowserRouter>
        <UpdateOrg />
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Check render elements UpdateOrg", () => {
    const { container } = render(
      <BrowserRouter>
        <UpdateOrg />
      </BrowserRouter>
    );

    const containerClass = container.querySelector(".container");
    expect(containerClass).toBeDefined();
  });

  test("Check UpdateOrg content", () => {
    const component = renderWithContext(
      <BrowserRouter>
        <UpdateOrg />
      </BrowserRouter>
    );

    const valueElement = component.root.find(
      (el) => el.type === "b" && el.props.className === "b1"
    );

    console.log(valueElement.props);

    expect(valueElement.props.children).toEqual("Upload JSON here!!");
  });
});
