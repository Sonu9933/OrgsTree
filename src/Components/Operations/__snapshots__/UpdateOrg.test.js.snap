// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Org Should render UpdateOrg 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto text-center form p-4"
    >
      <h1
        className="text-center display-4 py-2 text-truncate text-primary"
      >
        <b
          className="b1"
        >
          Upload JSON
        </b>
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-8 col-md-8 col-sm-10 mx-auto text-center form p-4"
    >
      <h4
        className="display-4 py-2 border border-primary text-info rounded"
      >
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <input
            autoComplete="off"
            multiple={true}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            tabIndex={-1}
            type="file"
          />
          Drag 'n' drop some files here, or click to select files
        </div>
      </h4>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xl-5 col-lg-6 col-md-8 col-sm-10 mx-auto text-center form p-4"
    >
      Go back to Home
      <b>
        <a
          className="Info link px-2"
          href="/"
          onClick={[Function]}
        >
          Goto Home
        </a>
      </b>
    </div>
  </div>
</div>
`;
